modules/machineset-creating-efa-options.adoc

```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-aws.adoc
+
+:_content-type: PROCEDURE
+[id="machineset-creating-efa-options_{context}"]
+= Creating machines that use an AWS Elastic Fabric Adapter
+
+You can deploy compute machines that use an AWS Elastic Fabric Adapter (EFA) by adding the `networkInterfaceType` field to the machine set YAML file for your compute machines.
+
+.Prerequisites
+
+* You have configured security groups to support the use of an EFA.
++
+Machines that use an EFA must belong to security groups that allow all traffic between all hosts in the security group. Clusters that use the Machine API are not configured to meet this requirement by default.
++
+You can manually configure your security groups to support the use of an EFA by using the AWS Management Console or the AWS CLI. For more information, see the Amazon EC2 documentation about https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/working-with-security-groups.html#updating-security-group-rules[updating security group rules].
+
+.Procedure
+
+. Generate a base64-encoded string for a file that removes `memlock` limits.
++
+.Example unencoded data
+[source,terminal]
+----
+[crio.runtime]
+default_ulimits = [
+        "memlock=-1:-1"
+]
+----
++
+.Example base64-encoded data
+[source,terminal]
+----
+W2NyaW8ucnVudGltZV0KZGVmYXVsdF91bGltaXRzID0gWwogICAgICAgICJtZW1sb2NrPS0xOi0xIgpdCg==
+----
+
+. Create a file named `unlimited-memlock.yaml` that contains the following YAML definition:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 02-worker-container-runtime <1>
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,<base64-encoded-memlock-data> <2>
+        mode: 420
+        overwrite: true
+        path: /etc/crio/crio.conf.d/10-memlock <3>
+----
+<1> Specify a name for the machine configuration.
+<2> Specify a base64-encoded string for the unlimited `memlock` file data.
+<3> Specify the path for the `memlock` resource.
+
+. To create the `MachineConfig` object, enter the following command:
++
+[source,terminal]
+----
+$ oc create -f unlimited-memlock.yaml
+----
```

Comment:
When did testing, we can create machine with EFA networkInterfaceType without these steps, so I think we can move this part (creating unlimited-memlock) to modules/machineset-enabling-efa-options.adoc